### PR TITLE
Add MavenJavadocNonAsciiRecipe to remove non-ASCII characters from Ja…

### PR DESCRIPTION
**Pull request by sfarhaazmi**
…vadoc

## What's changed?
Added `MavenJavadocNonAsciiRecipe` to automatically remove non-ASCII characters from Javadoc comments. This recipe helps prevent Maven javadoc-plugin build failures caused by non-ASCII characters with ambiguous error messages.

## What's your motivation?
Maven's javadoc-plugin does not handle non-ASCII characters well and often fails with very ambiguous error messages that don't clearly indicate the root cause. This makes it difficult for developers to identify and fix the issue, especially in international codebases where accented characters, special symbols, or other non-ASCII content may be present in documentation.

## Anything in particular you'd like reviewers to focus on?
- **Recipe Logic**: The Unicode normalization approach using `Normalizer.Form.NFKD` followed by ASCII filtering
- **Test Coverage**: Verification that both positive cases (non-ASCII removal) and negative cases (ASCII preservation) work correctly
- **Performance**: The recipe processes comments efficiently without unnecessary tree traversals
- **Edge Cases**: Handling of empty/null text and various Unicode character types

### CODE REVIEW

File: src/test/java/org/openrewrite/staticanalysis/maven/MavenJavadocNonAsciiRecipeTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                * this is a sample class
+                * ₤€ contains non ascii characters
+                */
+                class A {
+                    /**
+                    * this is the main method
+                    */
+                    public static void main(String... args) {
+                        System.out.println("Hello World!");
+                    }
```
Comment (greg-at-moderne) [inline src/test/java/org/openrewrite/staticanalysis/maven/MavenJavadocNonAsciiRecipeTest.java:48]: The method could be removed totally as it doesn't add anything to the test.

**Review by greg-at-moderne — COMMENTED** [review]
Thanks for the contribution!

I suggest to shorten the existing tests to avoid excess.

Also, would you be so kind to add more test cases for things like:
- non-ASCII punctuation,
- non-ASCII whitespace,
- some typical texts in various languages/scripts
?

**Comment by sfarhaazmi** [comment]
Added some more tests for non ascii white space, punctuations, emojis and languages.

File: src/main/java/org/openrewrite/staticanalysis/maven/MavenJavadocNonAsciiRecipe.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    public class MavenJavadocNonAsciiPruner extends JavaIsoVisitor<ExecutionContext> {
+
+        @Override
+        public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, ExecutionContext ctx) {
+            return processComments(super.visitClassDeclaration(classDecl, ctx), classDecl.getComments());
+        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+            return processComments(super.visitMethodDeclaration(method, ctx), method.getComments());
+        public J.VariableDeclarations.NamedVariable visitVariable(J.VariableDeclarations.NamedVariable variable, ExecutionContext ctx) {
+            return processComments(super.visitVariable(variable, ctx), variable.getComments());
+            return processComments(super.visitMethodDeclaration(method, executionContext), method.getComments());
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/staticanalysis/maven/MavenJavadocNonAsciiRecipe.java:74]:
```suggestion
```

File: src/main/java/org/openrewrite/staticanalysis/maven/MavenJavadocNonAsciiRecipe.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+            return processComments(super.visitMethodDeclaration(method, ctx), method.getComments());
+        public J.VariableDeclarations.NamedVariable visitVariable(J.VariableDeclarations.NamedVariable variable, ExecutionContext ctx) {
+            return processComments(super.visitVariable(variable, ctx), variable.getComments());
+        }
+
+        @Override
+        public J.VariableDeclarations.NamedVariable visitVariable(J.VariableDeclarations.NamedVariable variable, ExecutionContext executionContext) {
+            return processComments(super.visitVariable(variable, executionContext), variable.getComments());
+        }
```
[the remark this answers is not in the document]
Comment (greg-at-moderne) [reply src/main/java/org/openrewrite/staticanalysis/maven/MavenJavadocNonAsciiRecipe.java]: ... and 4409dfb

**Comment by timtebeek** [comment]
Great to see! Couple quick questions: would it make sense to use `visitSpace` here instead?
Should we add this recipe to the Apache Maven best practices downstream in rewrite-apache?

**Comment by timtebeek** [comment]
Here's the reduced example; thanks both! https://github.com/openrewrite/rewrite-static-analysis/commit/6aaa59ef15de91434bc1355860a08ac5b67645c0

**Comment by sfarhaazmi** [comment]
> Here's the reduced example; thanks both! [6aaa59e](https://github.com/openrewrite/rewrite-static-analysis/commit/6aaa59ef15de91434bc1355860a08ac5b67645c0)

Thank you @timtebeek that looks great!
